### PR TITLE
Created search-view

### PR DESCRIPTION
**Pull request by cjmitchell44**
This is just a start on the search page, but most of the fundamentals are in place.

### CODE REVIEW

File: src/client/js/search.js [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+export class SearchView {
+  #searchType = "Trail";
+
+  constructor() {}
+
+  async render() {
+    // Create root element
+    const searchView = document.createElement("div");
+    searchView.id = "home-view";
```
Comment (angelikaladia) [inline src/client/js/search.js:11]: should be "search-view"

File: src/client/js/search.js [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+  }
+
+  async #getFlights() {
+    // Fake flight
+    const flight = new Flight("Flight 1");
+    return [flight];
+  }
+}
+
+class Trail {
```
Comment (angelikaladia) [inline src/client/js/search.js:220]: We should be keeping the definitions for class the same across all files. Refer to the definition of `Trail` in `existing-trips.js`

File: src/client/js/search.js [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+  name = null;
+
+  constructor(name) {
+    this.name = name;
+  }
+}
+
+class Hotel {
```
Comment (angelikaladia) [inline src/client/js/search.js:228]: Same as above. For Hotel and Flights, refer to the definition in `edit-trip.js`

**Review by angelikaladia — COMMENTED** [review]
I think it would be best to add a "go" button for uses to click instead of automatically populating the results once the type of search is put in. I know it won't do much right now but maybe just change it so that only the go button calls the get methods.

File: src/client/js/search.js [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+  constructor() {}
+
+  async render() {
+    // Create root element
+    const searchView = document.createElement("div");
+    searchView.id = "home-view";
+
+    // Create child elements
+    const headerImage = document.createElement("img");
+    headerImage.src = "./img/generic-background.png";
```
Comment (dominicrutk) [inline src/client/js/search.js:15]: Add an `alt` attribute to the image to make sure it is accessible

File: src/client/js/search.js [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+    this.#searchType = searchType;
+  }
+
+  render() {
+    // Create div for search fields
+    const searchInterface = document.createElement("div");
+
+    // Create dropdown for type of search
+    const typeLabel = document.createElement("label");
+    typeLabel.innerHTML = "Type: ";
```
Comment (dominicrutk) [inline src/client/js/search.js:61]: Labels should have a `for` attribute that matches the `id` of the form element they are labeling (in this case `searchType`)

File: src/client/js/search.js [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+
+  constructor(inputName, options, searchType) {
+    this.#inputName = inputName;
+    this.#options = options;
+  }
+
+  render() {
+    // Create a label and dropdown for a search selection
+    const label = document.createElement("label");
+    label.innerHTML = this.#inputName;
```
Comment (dominicrutk) [inline src/client/js/search.js:131]: Same comment about labels having a `for` attribute as above

File: src/client/js/search.js [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+  #fieldName = null;
+
+  constructor(fieldName) {
+    this.#fieldName = fieldName;
+  }
+
+  render() {
+    // Create a label and input for a search field
+    const label = document.createElement("label");
+    label.innerHTML = this.#fieldName;
```
Comment (dominicrutk) [inline src/client/js/search.js:155]: `for` attribute, again

File: src/client/js/search.js [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+    } else if (this.#searchType === "Flight") {
+      this.#results = await this.#getFlights();
+    }
+
+    // Add results to list
+    this.#results.forEach((result) => {
+      const listElement = document.createElement("div");
+      listElement.innerHTML = result.name;
+      results.appendChild(listElement);
+    });
```
Comment (dominicrutk) [inline src/client/js/search.js:196]: Don't use a `ul` element unless you're adding `li` elements to it, which you're not doing here. In general, `ul` elements only make sense for actual bulleted lists, but what we have here is more complex (a list of `div` elements), so just make the entire `results` element a `div`.

**Review by angelikaladia — APPROVED** [review]

File: src/client/js/search.js [hunk 9]
```diff
@@ -153,13 +161,38 @@ class SearchField {
     // Create a label and input for a search field
     const label = document.createElement("label");
     label.innerHTML = this.#fieldName;
+    label.for = this.#fieldName;
+
     const searchFieldElm = document.createElement("input");
+    searchFieldElm.id = this;
```
Comment (dominicrutk) [inline src/client/js/search.js:167]: Is this supposed to be `this.#fieldName`?

**Pull request merged**